packages/phone-number-privacy/signer/src/common/database/wrappers/request.ts

```diff
@@ -43,7 +43,10 @@ export async function insertRequest(
   return doMeteredSql('insertRequest', ErrorMessage.DATABASE_INSERT_FAILURE, logger, async () => {
     const sql = db<PnpSignRequestRecord>(REQUESTS_TABLE)
       .insert(toPnpSignRequestRecord(account, blindedQuery, signature))
+      .onConflict([REQUESTS_COLUMNS.address, REQUESTS_COLUMNS.blindedQuery])
+      .ignore()
```

Comment:
Perhaps you could just update the timestamp in the case of conflicts. I don't know if that has any value.